src/easyscience/fitting/calculators/interface_factory.py

```diff
@@ -174,7 +168,7 @@ def generate_bindings(self, model, *args, ifun=None, **kwargs):
                 prop._callback = item.make_prop(item_key)
                 prop._callback.fset(prop_value)
 
-    def __call__(self, *args, **kwargs) -> _M:
+    def __call__(self, *args, **kwargs) -> None:
```

Comment:
Usage of `_M` in this class makes NO SENSE.
This really should be
```python
_M = TypeVar('_M')

class InterfaceTemplate(Generic[_M]):
    def __init__(self):
        self.__interface_obj: Optional[_M] = None

    def __call__(self, *args, **kwargs) -> Optional[_M]:
        return self.__interface_obj
```
if we want the class to remember what `_M` is, or even better, skip `Optional` and assign later

```python
class InterfaceTemplate(Generic[_M]):
    def __init__(self, obj: _M):
        self.__interface_obj: _M = obj

    def __call__(self, *args, **kwargs) -> _M:
        return self.__interface_obj
```

Reply:
(mentioned because you replaced TypeVar with explicit types)